### PR TITLE
Add document for hive functions

### PR DESCRIPTION
**Pull request by yulongfufu**
Add document for #16737
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by rongrong** [comment]
General comments: The documentation should be about what the feature is and how to use it, rather than how it's implemented. So the content should include: What is hive UDF (can be external reference), what is supported and how to use them in Presto query. Please restructure accordingly. Thanks!

File: presto-docs/src/main/sphinx/functions/hive.rst [hunk 1]
```diff
@@ -0,0 +1,49 @@
+==============================
+Hive Functions
+==============================
+
+It is a way of reuse of the hive functions in presto. All of the hive
+function classes exist in presto or external storage. We encode / decode
+the parameters to transform the type between presto and hive, and then
+wrap the hive function as the presto function.
```
Comment (tdcmeehan) [inline presto-docs/src/main/sphinx/functions/hive.rst:8]:
```suggestion
Presto allows you to use the same UDFs that are available in your Hive deployment. All Hive UDFs may
exist in Presto or external storage. Parameters and return values are converted between the types expected in Presto
and Hive.
```

File: presto-docs/src/main/sphinx/functions/hive.rst [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+The Hive Functions has 3 components:
+
+* Hive Scalar Function
+* Hive Aggregation Function
+* Hive Window Function
+
+We support hive built-in scalar function now.
+
```
Comment (tdcmeehan) [inline presto-docs/src/main/sphinx/functions/hive.rst:17]:
```suggestion
Note: of the three supported Hive function types (scalar, windowing and aggregate functions), only scalar functions are currently supported.
```

File: presto-docs/src/main/sphinx/functions/hive.rst [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+Hive Scalar Functions
+-----------------
+
+.. note::
+
+    Hive has two different interfaces for writing Hive UDF.
+    One is the basic UDF interface (deprecated) and the other
+    is the GenericUDF interface.
+
+    We support hive built-in GenericUDf now.
```
Comment (tdcmeehan) [inline presto-docs/src/main/sphinx/functions/hive.rst:27]:
```suggestion
    Hive has two different interfaces for writing Hive UDF.
    One is the basic UDF interface (deprecated) and the other
    is the GenericUDF interface.

    We currently only support the Hive built-in GenericUDF.
```

**Review by tdcmeehan — CHANGES_REQUESTED** [review]

File: presto-docs/src/main/sphinx/functions/hive.rst (hunk 3 above)
Comment (yulongfufu) [reply presto-docs/src/main/sphinx/functions/hive.rst]: That's very nice of you!
I have modified the doc according to your comments and add some describes about the hive udf.

**Comment by aweisberg** [comment]
Hi! I added a commit with some changes since it got pretty involved. If you think they are good can you squash them and then I will approve?

**Comment by aweisberg** [comment]
Actually I have one question, UDFs can come from external storage right? How do you configure that? Should that be in the documentation?

**Comment by yulongfufu** [comment]
> UDFs` can come from external storage right?   How do you configure that?  Should that be in the documentation?

Yes. I have pushed the changes that get external storage functions logic into later Pr: `support external hive function`. Actually it just is another functionRegistry like `StaticHiveFunctionRegistry`. We can choose which one functionRegistry can be load by the `hive-function.properties`.
I want to contribute the  `"support external hive function"` after support hive aggregation function. So it will not be in this doc :)

**Review by aweisberg — APPROVED** [review]
LGTM, TY!

File: presto-docs/src/main/sphinx/functions/hive.rst [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
++---------------+-----------------------------------+-----------------------------------+
+| genericUdf    | str_to_map                        | facebook_str_to_map               |
++---------------+-----------------------------------+-----------------------------------+
+
+See the `Hive Language Manual <https://cwiki.apache.org/confluence/display/hive/languagemanual+udf>`_ for all hive scalar functions information.
+
+Examples
+--------
+
+Use hive function namespace ``hive.default`` to call the hive function.
```
Comment (rongrong) [inline presto-docs/src/main/sphinx/functions/hive.rst:41]: Is this plugin enabled by default or do people need to set it up? If people need to setup the Hive function namespace to use it, please also document how to enable it.

Comment (yulongfufu) [reply presto-docs/src/main/sphinx/functions/hive.rst:47]: I added the description of the Hive function namespace plugin.

**Comment by tdcmeehan** [comment]
@yulongfufu the information here would be great to merge back into trunk.  Is this waiting review still?

**Comment by steveburnett** [comment]
> @yulongfufu the information here would be great to merge back into trunk. Is this waiting review still?

Commenting to raise awareness of this PR - what can be done to make progress on this?

**Comment by steveburnett** [comment]
Closing due to lack of activity and age. If this is still relevant, please re-open this PR, rebase, and resolve any file conflicts.

**Pull request closed without merging**